Externalize item graphics

selaux opened the pull request:
This changes externalized data, so images for items can be changed more easily. Before `ubGraphicType` and `ubGraphicNum` were used to determine in code which file to use for small/big inventory graphics as well as tile graphics. Now these graphics are referenced explicitly by the `inventoryGraphics` and `tileGraphic` properties. This allows them to be changed independent of each other and without having to overwrite the original item graphics.

I tried to implement the `inventoryGraphics` property as robust as possible. So when one of the images is not found, it simply shows a fallback image (see screenshots). The fallback images were created using free-ja2-resources here: https://github.com/ja2-stracciatella/free-ja2-resources/pull/8


![Screenshot from 2022-01-21 12-00-55](https://user-images.githubusercontent.com/848854/150516310-76b0df54-4546-494d-af87-f439fa649fd1.png)
![Screenshot from 2022-01-21 12-01-00](https://user-images.githubusercontent.com/848854/150516322-09461e1a-8657-4342-9996-e9496e0e03d3.png)

lynxlynxlynx commented on src/externalized/DefaultContentManager.cc:
```diff
@@ -1190,6 +1190,21 @@ const ItemModel* DefaultContentManager::getItemByName(const ST::string &internal
+		auto& path = item->getInventoryGraphicSmall().getPath();
+		auto existing = std::find(v.begin(), v.end(), path);
+		if (existing == v.end()) {
+			v.push_back(item->getInventoryGraphicSmall().getPath());
```
Why not just push `path`?

selaux replied:
Good point.

lynxlynxlynx commented on src/externalized/InventoryGraphicsModel.cc:
```diff
@@ -0,0 +1,21 @@
+    v.AddMember("big", big);
+
+    return v;
+}
```
Didn't we standardize on tabs? Either way line 9 is misaligned.

selaux replied:
Yep we did. But my editor didn't know. Reindented the new files with tabs.

ja2-derek replied:
Maybe time for an EditorConfig!

lynxlynxlynx commented:
Other than that, LGTM. But I didn't double check all the jsons. Is there any particular reason you made `small` an object and not just adding everything to the same level as `big`?

selaux commented:
The JSONs were adapted programatically, so I don't think there should be anything broken in there.

You have a good point about the `small` and `large` graphic variants. Originally in the game the `small` variant was always an STI with images for multiple items, while the `big` variant only had a single image. That is why the structure is was as you described. But I don't see why we should enforce that. Let people use whatever images they choose. I changed it to have the same structure.

I also changed a little bit of stuff in `Interface_Items` again, so images which have an out-of-bound `subImageIndex` are handled gracefully as well.

selaux commented:
Adapted it once more to match https://github.com/ja2-stracciatella/ja2-stracciatella/pull/1460
